functorch/functorch/_src/aot_autograd.py

```diff
@@ -244,6 +261,14 @@ def fake_fn(primals, tangents):
                 with track_graph_compiling("forward"):
                     compiled_fw = fw_compiler(fw_module, flat_tensor_args)
                 fw_outs = normalize_as_list(compiled_fw(*flat_tensor_args))
+                if config.use_functionalize:
+                    # Call the input mutation submodule with the mutated inputs
+                    mutated_inputs = fw_outs[:num_mutated_inputs]
```

Comment:
It seems like we're assuming that only the forwards pass will have its inputs mutated? Is this true?

Reply:
Yep (I commented that assumption somewhere else in this file). I was talking to @soulitzer a few weeks ago about it, and it sounds like it shouldn't be valid for the backward to mutate its program inputs (?).

Reply:
makes sense